YoCode/FrontEndCheck.cs

```diff
@@ -5,60 +5,98 @@
             {
                 try
                 {
-                    var foxService = FirefoxDriverService.CreateDefaultService(Directory.GetCurrentDirectory());
-                    foxService.HideCommandPromptWindow = true;
+                    service = FirefoxDriverService.CreateDefaultService(Directory.GetCurrentDirectory());
+                    service.HideCommandPromptWindow = true;
                     port = applicantsWebPort;
                     var options = new FirefoxOptions();
                     options.AddArgument("--headless");
-
-                    browser = new FirefoxDriver(foxService, options);
+                    browser = new FirefoxDriver((FirefoxDriverService)service, options);
                 }
-                catch (Exception)
+                catch { };
+            }
+            else if (browsers.Any(a=>a.Contains(CHROME)))
+            {
+                try
                 {
-                    var chromeService = ChromeDriverService.CreateDefaultService(Directory.GetCurrentDirectory());
-                    chromeService.HideCommandPromptWindow = true;
+                    service = ChromeDriverService.CreateDefaultService(Directory.GetCurrentDirectory());
+                    service.HideCommandPromptWindow = true;
                     port = applicantsWebPort;
                     var chromeOptions = new ChromeOptions();
                     chromeOptions.AddArgument("--headless");
 
-                    browser = new ChromeDriver(chromeService, chromeOptions);
+                    browser = new ChromeDriver((ChromeDriverService)service, chromeOptions);
                 }
+                catch { };
+            }
+            else
+            {
+                FrontEndEvidence.SetFailed($"Could not execute check: Did not find needed browser{Environment.NewLine}Please install Google Chrome or Mozilla Firefox internet browser");
+            }
 
+            try
+            {
                 browser.Navigate().GoToUrl(port);
 
                 FrontEndEvidence.FeatureImplemented = CheckIfUIContainsFeature(keyWords);
 
                 UIKeywords.GARBAGE_INPUT.ToList().ForEach(InputData);
+            }
+            catch { return; }
 
-                if (!FrontEndEvidence.Evidence.Any())
-                {
-                    FrontEndEvidence.GiveEvidence("Could not input any data");
-                }
+            if (!FrontEndEvidence.Evidence.Any())
+            {
+                FrontEndEvidence.GiveEvidence("Could not input any data");
+            }
 
+            CloseBrowser();
+        }
+
+        public static bool CloseBrowser()
+        {
+            if (browser != null)
+            {
+                Console.WriteLine("browser closed");
                 browser.Dispose();
+                browser.Quit();
+                running = false;
+                return true;
             }
-            catch (WebDriverException e)
+            if(running)
```

Comment:
The `running` variable is a boolean. When you can it's better to use boolean values directly. So rather than:

```
if(running) 
{
    return true;
} else
{
    return false;
}
```
You just need to:

```
return running;
```